managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1729,6 +1729,18 @@ synchronized void ledgerClosed(final LedgerHandle lh) {
         }
     }
 
+    @Override
+    public void removeNonRecoverableLedger(long ledgerId){
+        ledgers.remove(ledgerId);
+        Iterator<ManagedCursor> managedCursorIterator = cursors.iterator();
+        while (managedCursorIterator.hasNext()){
+            ManagedCursor managedCursor = managedCursorIterator.next();
+            if (managedCursor instanceof ManagedCursorImpl managedCursorImpl){
+                managedCursorImpl.clearIncompleteAckedRecordsByLedgerId(ledgerId);
```

Comment:
Using interface implementations may break encapsulation, but I'm not sure it's justified here.

Reply:
fixed